src/providers/walletconnectNotify.ts

```diff
@@ -0,0 +1,169 @@
+  const subscribersFromWalletConnect = await getSubscribersFromWalletConnect();
+
+  // optimistically reserve all subscribers from the db
+  const crossReferencedSubscribers = new Array(subscribersFromDb.length);
+
+  // Create a hashmap for faster lookup
+  const addressPrefixMap = new Map<string, string>();
+  for(const subscriber of subscribersFromWalletConnect) {
+    const unprefixedAddress = subscriber.split(':').pop();
+    if(unprefixedAddress) {
+      addressPrefixMap.set(unprefixedAddress, subscriber)
+    }
+  }
+
+  for(const subscriber of subscribersFromDb) {
+    const crossReferencedAddress = addressPrefixMap.get(subscriber);
+    if(crossReferencedAddress) {
+      crossReferencedSubscribers.push(crossReferencedAddress)
+    }
+  }
+
+  // remove empty elements from the array, since some might not have been found in WalletConnect Notify server
+  return crossReferencedSubscribers.filter(addresses => addresses);
+}
+
+async function queueNotificationsToSend(notification, accounts: string[]) {
+  for(let i = 0; i < accounts.length; i += MAX_ACCOUNTS_PER_REQUEST) {
+    await sendNotification(notification, accounts.slice(i, i + MAX_ACCOUNTS_PER_REQUEST))
+    const waitTime = (1 / PER_SECOND_RATE_LIMIT) + WAIT_ERROR_MARGIN;
+    await wait(waitTime); 
+  }
+}
+
+export async function sendNotification(notification, accounts) {
+  const notifyUrl = `${WALLETCONNECT_NOTIFY_SERVER_URL}/${WALLETCONNECT_PROJECT_ID}/notify`;
+
+  const body = {
+    accounts,
+    notification
+  };
+
+  try {
+    const notifyRs = await fetch(notifyUrl, {
+      method: 'POST',
+      headers: {
+        ...AUTH_HEADER,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(body)
+    });
+
+    const notifySuccess = await notifyRs.json();
+
+    return notifySuccess;
+  } catch (e) {
+    capture('[WalletConnect] failed to notify subscribers', e);
+  }
+}
+
+// Transform proposal event into notification format.
+async function formatMessage(event, proposal) {
+  const space = proposal.space;
+  if (!space) return null;
+
+  const notificationType = getNotificationType(event.event);
+  const notificationBody = getNotificationBody(event.event, space);
+
+  if (!notificationType) {
+    capture(`[WalletConnect] could not get matching notification type for event ${event.event}`);
+    return;
+  }
+
+  if(!notificationBody) {
+    capture(`[WalletConnect] could not get matching notification body for event ${event.event}`);
+    return;
+  }
+
+  const url = new URL(proposal.link);
+  url.searchParams.append('app', 'walletconnect')
+
+  return {
+    title: proposal.title,
+    body: notificationBody,
+    url: url.toString(),
+    icon: space.avatar,
+    type: notificationType
+  };
+}
+
+export async function send(event, proposal, _subscribers) {
+  
+  const crossReferencedSubscribers = await crossReferenceSubscribers(proposal.space);
+  const notificationMessage = await formatMessage(event, proposal);
```

Comment:
This function is not a promise, you can remove `await`
```suggestion
  const notificationMessage = formatMessage(event, proposal);
```